File: client/src/services/ProfileService.js
import { AppState } from "../AppState.js"
import { Keep } from "../models/Keep.js"
import { Profile } from "../models/Profile.js"
import { Vault } from "../models/Vault.js"
import { logger } from "../utils/Logger.js"
import { api } from "./AxiosService.js"


class ProfileService{

    async getProfile(profileId){
        const response = await api.get(`api/profiles/${profileId}`)
        const profile = new Profile(response.data)
        AppState.activeProfile = profile
    }

    async getVaultsByProfile(profileId){
        const response = await api.get(`api/profiles/${profileId}/vaults`)
        logger.log(response.data)
        const vaults = response.data.map(vaultData => new Vault(vaultData))
        logger.log(vaults)
        AppState.vaults = vaults
    }
    
    async getKeepsByProfile(profileId){
        const response = await api.get(`api/profiles/${profileId}/keeps`)
        const keeps = response.data.map(keepData => new Keep(keepData))
        AppState.keeps = keeps
    }

}

export const profileService = new ProfileService()

File: server/Services/ProfileService.cs


namespace keepr.Services;

public class ProfileService
{
    private readonly ProfileRepository _repository;
    private readonly VaultService _vaultService;
    private readonly KeepService _keepService;
    private readonly VaultKeepService _vaultkeepService;
    public ProfileService(ProfileRepository repository, VaultKeepService vaultkeepService, KeepService keepService, VaultService vaultService)
    {
        _repository = repository;
        _vaultkeepService = vaultkeepService;
        _keepService = keepService;
        _vaultService = vaultService;
    }

    internal List<KeptVaultKeep> GetKeepsByProfile(string profileId)
    {
        List<KeptVaultKeep> keeps = _keepService.GetKeepsByProfile(profileId);
        return keeps;
    }

    internal Profile GetProfileById(string profileId)
    {
        Profile profile = _repository.GetProfileById(profileId);
        return profile;
    }

    internal List<Vault> GetVaultsByProfile(string profileId)
    {
        List<Vault> vaults = _vaultService.GetPublicVaultsByProfile(profileId);
        return vaults;
    }
}